File: 35-SearchInsertPosition/35-SearchInsertPosition.java
// Last updated: 2/27/2026, 5:12:11 PM
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        int i = 0;
        int j = nums.length - 1;
        int mid = i + ( j - i ) / 2;

        while(i <= j){

            if(target == nums[mid]) return mid;

            else if(target < nums[mid]) j = mid - 1;
            else if(target > nums[mid]) i = mid + 1;

            mid = i + (j - i) / 2;
        }

        return j + 1;
        


    }
}

File: 35-SearchInsertPosition/35-SearchInsertPosition.py
# Last updated: 7/12/2026, 6:25:06 PM
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:

        n = len(nums)
        l , r = 0, n - 1

        while l <= r:
            mid = (l + r) // 2

            if nums[mid] >= target:
                r = mid - 1
            else:
                l = mid + 1
        
        return l

        